### PR TITLE
[Mellanox] add x86_64-nvidia_sn4280_simx-r0 platform

### PR DESCRIPTION
**Pull request by Yakiv-Huryk**
#### Why I did it
To add a new simulation platform SN4280-simx

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Added/updated all relevant files for a new device

#### How to verify it
Run Nvidia simulation on SN4280 (ASIC and Platform)

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by vivekrnv — APPROVED** [review]

**Review by dgsudharsan — APPROVED** [review]

**Review by liat-grozovik — CHANGES_REQUESTED** [review]

File: device/mellanox/x86_64-nvidia_sn4280_simx-r0/ACS-SN4280/buffers_defaults_t0.j2 [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{#
```
[the remark this answers is not in the document]
Comment (Yakiv-Huryk) [reply device/mellanox/x86_64-nvidia_sn4280_simx-r0/ACS-SN4280/buffers_defaults_t0.j2]: made it a link to SN4700

File: device/mellanox/x86_64-nvidia_sn4280_simx-r0/ACS-SN4280/buffers_defaults_t1.j2 [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{#
```
[the remark this answers is not in the document]
Comment (Yakiv-Huryk) [reply device/mellanox/x86_64-nvidia_sn4280_simx-r0/ACS-SN4280/buffers_defaults_t1.j2]: made it a link to SN4700

**Review by liat-grozovik — APPROVED** [review]

File: device/mellanox/x86_64-nvidia_sn4280_simx-r0/ACS-SN4280/hwsku.json [hunk 3]
```diff
@@ -0,0 +1,100 @@
+{
```
[the remark this answers is not in the document]
Comment (Yakiv-Huryk) [reply device/mellanox/x86_64-nvidia_sn4280_simx-r0/ACS-SN4280/hwsku.json:1]: 32 for now, once the x86_64-nvidia_sn4280-r0 is defined, the SKU will be a link to x86_64-nvidia_sn4280-r0/ACS-SN4280 (as it's done for other simx platforms)

File: device/mellanox/x86_64-nvidia_sn4280_simx-r0/ACS-SN4280/sai_4280.xml [hunk 4]
```diff
@@ -0,0 +1,265 @@
+<?xml version="1.0"?>
```
[the remark this answers is not in the document]
Comment (Yakiv-Huryk) [reply device/mellanox/x86_64-nvidia_sn4280_simx-r0/ACS-SN4280/sai_4280.xml:1]: fixed

File: device/mellanox/x86_64-nvidia_sn4280_simx-r0/pcie.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
+##
```
[the remark this answers is not in the document]
Comment (Yakiv-Huryk) [reply device/mellanox/x86_64-nvidia_sn4280_simx-r0/pcie.yaml]: fixed

**Pull request merged**